Assets/Scripts/UI/Achievements/UIAchievementsMenu.cs

```diff
@@ -134,48 +131,35 @@ public void HideMenu()
             achievementLockedIcon.gameObject.SetActive(false);
         }
 
-        public void IncrementLoginStat()
+        public async void IncrementLoginStat()
         {
-            var statsInterface = EOSManager.Instance.GetEOSPlatformInterface().GetStatsInterface();
-            var userId = EOSManager.Instance.GetProductUserId();
-            IngestStatOptions ingestOptions = new IngestStatOptions()
-            {
-                LocalUserId = userId,
-                TargetUserId = userId,
-                Stats = new IngestData[] { new IngestData() { StatName = "login_count", IngestAmount = 1 } }
-            };
-
-            statsInterface.IngestStat(ref ingestOptions, null, (ref IngestStatCompleteCallbackInfo info) =>
-            {
-                Debug.LogFormat("Stat ingest result: {0}", info.ResultCode.ToString());
-                achievementManager.Refresh();
-            });
+            await StatsManager.Instance.IngestStatAsync("login_count", 1);
         }
 
         //manually unlock achievement being displayed
         //TODO: refresh achievement data without having to log out
-        public void UnlockAchievement()
+        public async void UnlockAchievement()
         {
             if (displayIndex < 0 || displayIndex > EOSAchievementManager.GetAchievementsCount())
             {
                 return;
             }
 
-            var definition = achievementManager.GetAchievementDefinitionAtIndex(displayIndex);
+            var definition = EOSAchievementManager.Instance.GetAchievementDefinitionAtIndex(displayIndex);
 
-            achievementManager.UnlockAchievement(definition.AchievementId, (ref OnUnlockAchievementsCompleteCallbackInfo info) =>
+            try
             {
-                if (info.ResultCode == Result.Success)
-                {
-                    Debug.Log("UnlockAchievement Succeed"); 
-                    achievementManager.Refresh();
-                }
-            });
+                await EOSAchievementManager.Instance.UnlockAchievementAsync(definition.AchievementId);
+            }
+            catch (Exception e)
+            {
+                Debug.Log(e.Message);
```

Comment:
Should this either use the print method, or perhaps log at error or warning?

Reply:
The `UIAchievementsMenu` class does not use the `print` function. I did however update it to log an error instead of just a standard log.